QEP-auxiliary-storage.rst

```diff
@@ -0,0 +1,178 @@
+in a layer. Such auxiliary data are data used mostly for the needs of QGIS (symbology) and have no real
+interest in being stored with the native raw geospatial data.
+
+The need arises from the restrictions existing in the manual placement of labels.
+Manual placement of labels are possible in QGIS by setting some labeling properties (X and Y position,
+and rotation angle optionaly) as being "data-defined", meaning that values come from a column (or an expression).
+But setting this up on an existing layer requires either to add new columns to the source layer, while it is
+not always possible or desirable. Creating another table and using joins to it is possible but the synchronisation
+between the two tables is not possible, leading to a solution not satisfactory.
+
+We propose here a more integrated solution where the creation of auxiliary columns or layers
+are brought more transparently to the user by storing them in a central repository carried by the
+project file.
+
+Requirements
+------------
+
+The proposed solution have the following requirements :
+
+- It should allow to transparently add columns of any type to a layer, without adding them to the source data source.
+Beside usual types (string, real, integer), we may also think of geometry types. We only consider types that can be converted to atomic types (string for WKT or WKB)
+The user should be able to edit values in these columns when the layer is in editing mode.
+
+- These values should not be stored with the source data, and should be carried by the project, i.e.
+when the project file is moved or copied, the auxiliary data should still be there.
+
+- Adding a new row to the layer also adds a new row for each auxiliary field (with a null or default value)
+
+- Removing a row from the layer also removes the corresponding row of each auxiliary field
+
+- New GUI elements should ease the creation of auxiliary fields
+  - on each "data-defined" property
+  - possibly on groups of "data-defined": for manual label placement or label connectors
+
+- It should be possible to save auxiliary data to a new layer
+
+Proposed solution
+-----------------
+
+We propose to add a new type of field, close to the existing "virtual field" (where a field can be defined as an expression).
+Here a new type of field, "auxiliary field", will point to a central data storage.
+
+The central data storage should allow to store any kind of auxiliary column definition and data for any layer. And it should
+be linked to, or be part of, a project file. The access to auxiliary data should be efficient.
+
+Core changes
+------------
+
+Project file
+------------
+
+The current project file format (XML) is too limited to carry an efficient storage of such auxiliary data. It should somehow be evolved
+to include a binary indexable format. SQLite is chosen as a first candidate for this data format. But it might evolve if performance issues
+are found.
+
```

Comment:
Also, where will the sqlite file be stored while a project is being edited? I assume the user will be working with a temporary version of the sqlite database which is only stored inside the project zip on project save?